src/qgraphicsmozview_p.cpp

```diff
@@ -937,27 +927,61 @@ void QGraphicsMozViewPrivate::touchEvent(QTouchEvent* event)
         }
     }
 
-    if (meventStart.mTouches.Length()) {
-        // We should append previous touches to start event in order
-        // to make Gecko recognize it as new added touches to existing session
-        // and not evict it here http://hg.mozilla.org/mozilla-central/annotate/1d9c510b3742/layout/base/nsPresShell.cpp#l6135
-        if (meventMove.mTouches.Length()) {
-            meventStart.mTouches.AppendElements(meventMove.mTouches);
+    // We should append previous touches to start event in order
+    // to make Gecko recognize it as new added touches to existing session
+    // and not evict it here http://hg.mozilla.org/mozilla-central/annotate/1d9c510b3742/layout/base/nsPresShell.cpp#l6135
+    QList<int> startIds(moveIds);
+
+    // Produce separate event for every pressed touch points
+    Q_FOREACH (int id, pressedIds) {
+        MultiTouchInput meventStart(MultiTouchInput::MULTITOUCH_START, timeStamp, TimeStamp(), 0);
```

Comment:
While the ```meventStart``` existed previously I'd take a change now and add commit on top for renaming it.

Maybe something like ```multiTouchInputStart``` instead of ```meventStart``` to improve code readability.

Same of ```meventEnd``` and ```meventMove```